conn.go

```diff
@@ -39,70 +38,78 @@ type Conn interface {
 	Rooms() []string
 }
 
-type errorMessage struct {
-	namespace string
-	error
-}
-
 type writePacket struct {
 	header parser.Header
-	data   []interface{}
+
+	data []interface{}
 }
 
 type conn struct {
 	engineio.Conn
-	encoder    *parser.Encoder
-	decoder    *parser.Decoder
-	errorChan  chan errorMessage
-	writeChan  chan writePacket
-	quitChan   chan struct{}
+
+	id         uint64
 	handlers   map[string]*namespaceHandler
 	namespaces map[string]*namespaceConn
-	closeOnce  sync.Once
-	id         uint64
+
+	encoder *parser.Encoder
+	decoder *parser.Decoder
+
+	errorChan chan error
+	writeChan chan writePacket
+	quitChan  chan struct{}
+
+	closeOnce sync.Once
 }
 
-func newConn(c engineio.Conn, handlers map[string]*namespaceHandler) (*conn, error) {
+func newConn(c engineio.Conn, handlers map[string]*namespaceHandler) error {
 	ret := &conn{
 		Conn:       c,
 		encoder:    parser.NewEncoder(c),
 		decoder:    parser.NewDecoder(c),
-		errorChan:  make(chan errorMessage),
+		errorChan:  make(chan error),
 		writeChan:  make(chan writePacket),
 		quitChan:   make(chan struct{}),
 		handlers:   handlers,
 		namespaces: make(map[string]*namespaceConn),
 	}
+
 	if err := ret.connect(); err != nil {
-		ret.Close()
-		return nil, err
+		_ = ret.Close()
```

Comment:
why add `_ = `?

Reply:
in current master version doesn't catch a lot of inner mistakes :( i suggest in current version add underscore because it is simple way to search it in this project